### DISCUSSION TITLE
SOLVED: Is there a way to get a bounding box for set of (2d) objects

### DISCUSSION THREAD

**Discussion by hvegh** [BelfrySCAD/BOSL2#1363] [Q&A]
Is there a BOSL2 function that lets us:
- compute the bounding box,
- and the translation vector of this bounding box,
for an arbitrary collection of objects?

**SOLVED:
echo(pointlist_bounds(flatten(holes)));**

code below should give something like:

```
include <BOSL2/std.scad>;

holes = [
    circle(d=3),
    move([-10, 5], rect(2)),
    move([-10,-5], rect(2))
];
region(holes);
echo(pointlist_bounds(flatten(holes)));

// returns ECHO: [[-11, -6], [1.5, 6]]
```

![bbox](https://github.com/BelfrySCAD/BOSL2/assets/4540127/e4bdd626-67ae-48a0-8f90-b534a2ac415d)

**Comment by revarbat** [comment]
Sounds like you're looking for `pointlist_bounds()`

**Reply by revarbat** [reply]
Though you may need to use `flatten()` on the region before passing it to `pointlist_bounds()`.

**Comment by hvegh** [comment]
Great that did the trick, 
Thank you!
